### ISSUE TITLE
Schema validation is skipped for POST request if no body is present

### ISSUE THREAD

**Issue by mccare** [fastify/fastify#3866]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.29.0

### Node.js version

16.15.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

12.3.1

### Description

My schema validation is requiring a body with certain properties. When submitting an empty body the schema validation is skipped and the handler is called.

### Steps to Reproduce

See example repository here:
https://github.com/mccare/fastify-schema-validation-skipped.git

Basically handler with:
```
   fastify.post('/', {
        schema: {
            body: {
                properties: {
                    hello: { type: 'string' }
                },
                required: [ 'hello' ]
            }
        }
    }, async function (request, reply) {
        fastify.log.info('Handling request')
        return { root: true }
    })
```
and then call 
` curl -X POST http://localhost:3000`

### Expected Behavior

the curl call should produce a 400 bad request answer and not invoke my handler.

**Comment by climba03003** [comment]
I see the problem now .

You should either use a `shorten` form of schema or `full` schema. Not a `partial` schema.

```js
// Shorten Schema
fastify.post(
  "/",
  {
    schema: {
      body: {
        properties: {
          hello: { type: "string" },
        },
        required: ["hello"],
      },
    },
  },
  async function (request, reply) {
    fastify.log.info("Handling request");
    return { root: true };
  }
);

// Full Schema
fastify.post(
  "/",
  {
    schema: {
      body: {
        type: "object",
        properties: {
          hello: { type: "string" },
        },
        required: ["hello"],
      },
    },
  },
  async function (request, reply) {
    fastify.log.info("Handling request");
    return { root: true };
  }
);
```

**Comment by mccare** [comment]
Perfect, thanks for the clarification! Works with the full schema, shorten schema does not make the property required.
